edx_notifications/lib/tests/test_lib.py

```diff
@@ -0,0 +1,191 @@
+"""
+Tests for the publisher.py and consumer.py methods
+"""
+
+from django.test import TestCase
+
+from edx_notifications.lib.publisher import (
+    publish_notification_to_user,
+    register_notification_type
+)
+
+from edx_notifications.lib.consumer import (
+    get_notifications_count_for_user,
+    get_notifications_for_user,
+    mark_notification_read
+)
+
+from edx_notifications.data import (
+    NotificationMessage,
+    NotificationType,
+    NotificationUserMap,
+)
+
+from edx_notifications.exceptions import (
+    ItemNotFoundError,
+)
+
+
+class TestPublisherLibrary(TestCase):
+    """
+    Go through exposed endpoints in the publisher library
+    """
+
+    def setUp(self):
+        """
+        Initialize some data
+        """
+
+        self.test_user_id = 1001  # some bogus user identifier
+        self.msg_type = NotificationType(
+            name='open-edx.edx_notifications.lib.tests.test_publisher'
+        )
+        register_notification_type(self.msg_type)
+
+    def test_multiple_type_registrations(self):
+        """
+        Make sure the same type can be registered more than once
+        """
+
+        # msg_type.name is a primary key, so verify this does
+        # not throw an exception
+        register_notification_type(self.msg_type)
+
+    def test_publish_notification_to_user(self):
+        """
+        Go through and set up a notification and publish it
+        """
+
+        msg = NotificationMessage(
+            namespace='test-runner',
+            msg_type=self.msg_type,
+            payload={
+                'foo': 'bar'
+            }
+        )
```

Comment:
If message creation were a method on the `NotificationType`, then it could enforce that the required and optional arguments in the payload were passed correctly.